### DISCUSSION TITLE
Approved by...?

### DISCUSSION THREAD

**Discussion by nunodonato** [cjmellor/approval#48] [Q&A]
Hi!
I've been hunting for a package to handle approvals in models and this fits the needs almost perfectly!
However, in my use case, I also need to track *who* approved/rejected the change.

There doesn't seem to be a way to do this in this package, am I right?
I think I could use the events and get the currently authenticated user when it gets fired, but that will probably not work 100% of the time. Ideally, we would be able to pass that info to the approve() method.

Any suggestions?

**Comment by nunodonato** [comment]
Just noticed that the rolledback event constructor has all the useful info for this

```php
public function __construct(
        public Model $approval,
        public Authenticatable|null $user,
    )
```

Why wouldn't the other events have the same info available? It can be useful in a lot of scenarios

**Reply by cjmellor** [reply]
Hey,

I'd say the reason why the events don't send that data is purely because at the time, I didn't feel it was relevant to have it in, but, I'm very to open to have it in there, so you're welcome to do for a PR for it and I'll merge it (be sure to write tests). 

If you're not comfortable doing that, I can look into it myself but I can't say for sure when that will happen.

**Comment by cjmellor** [comment]
This was added in release 1.4.3 via #51
